File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-2.java
class Solution {
    private PriorityQueue<Integer> minHeap = new PriorityQueue<>();

    public int findKthLargest(int[] nums, int k) {
        for (var num : nums) {
            minHeap.offer(num);
            if (minHeap.size() > k) {
                minHeap.poll();
            }
        }

        return minHeap.peek();
    }
}

File: Data Structures & Algorithms/sort-array-by-parity/submission-2.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        n = len(nums)
        r, l = 0, n - 1
        result = [0] * n

        for num in nums:
            if num % 2 == 0:
                result[r] = num
                r += 1
            else:
                result[l] = num
                l -= 1

        return result